### DISCUSSION TITLE
How to ensure addCollection order?

### DISCUSSION THREAD

**Discussion by JaydeepGo** [11ty/eleventy#2341] [Q&A] [answered]
Hi, 

I am trying to add two customer collections like this.
```js
eleventyConfig.addCollection('posts', require('./src/_11ty/getAllPostsCollection'));
eleventyConfig.addCollection('tagList', require('./src/_11ty/getTagList'));
```

Here tagList function depends on posts collection but tagList collection is getting processed first. 

Is there any way to ensure posts collection gets processed first?

**Comment by pdehaan** [comment, marked as answer]
I don't think there is a way to guarantee ordering. But we might be able to come up with a solution/workaround if you can post the contents of the the `./src/_11ty/getAllPostsCollection.js` and `./src/_11ty/getTagList.js` files.

**Reply by JaydeepGo** [reply]
Hi @pdehaan  

### getAllPostsCollection.js
```js
const externalPosts = require('../_data/external-posts.js');

module.exports = function (collection) {
  let local_post = collection.getFilteredByGlob('./src/posts/*.md');
  let overallPosts = [...local_post, ...externalPosts];

  let posts = overallPosts.sort(function (a, b) {
    return b.date - a.date;
  });

  return [...posts];
};

```

### getTagList.js
``` js 
const excludedTags = ['all', 'nav', 'local-post', 'posts'];

module.exports = function (collection) {
  let tagSet = new Set();
  collection.getFilteredByTag('posts').forEach(function (item) {
    if ('tags' in item.data) {
      let tags = item.data.tags.filter((item) => !excludedTags.includes(item));

      for (const tag of tags) {
        tagSet.add(tag);
      }
    }
  });

  return [...tagSet];
};
```

**Reply by pdehaan** [reply]
Alright... I haven't tried rebuilding a version of this locally yet, but off the top of my head something like this might work:

```diff
 // src/_11ty/getTagList.js
+const allPostsCollection = require('./getAllPostsCollection');
 const excludedTags = ['all', 'nav', 'local-post', 'posts'];

 module.exports = function (collection) {
   let tagSet = new Set();
-  collection.getFilteredByTag('posts').forEach(function (item) {
+  allPostsCollection.forEach(function (item) {
     if ('tags' in item.data) {
       let tags = item.data.tags.filter((item) => !excludedTags.includes(item));

       for (const tag of tags) {
         tagSet.add(tag);
       }
     }
   });

   return [...tagSet];
 };
```

TLDR: Instead of rolling the dice on execution order, we just include the results from the `getTagList.js` file/collection. Sadly that means we'll execute `collection.getFilteredByGlob('./src/posts/*.md')` twice, plus whatever logic might be in `../_data/external-posts.js`, but 🤷 .

**Reply by JaydeepGo** [reply]
@pdehaan Thanks for your input.

**Reply by JaydeepGo** [reply]
Hi @zachleat is there any future plan to ensure addCollection order?
